### PR TITLE
[DO NOT MERGE] V11

### PR DESCRIPTION
**Pull request by mcardosos**
Thank you for your contribution to Go-AutoRest! We will triage and review it as soon as we can.

As part of submitting, please make sure you can make the following assertions:
 - [ ] I've tested my changes, adding unit tests if applicable.
 - [ ] I've added Apache 2.0 Headers to the top of any new source files.
 - [ ] I'm submitting this PR to the `dev` branch, except in the case of urgent bug fixes warranting their own release.
 - [ ] If I'm targeting `master`, I've updated [CHANGELOG.md](https://github.com/Azure/go-autorest/blob/master/CHANGELOG.md) to address the changes I'm making.

### CODE REVIEW

**Comment by marstr** [comment]
Have we updated any samples to use the `v11` branch to test the changes you've made against the services? If so, could you link us those changes?

**Comment by marstr** [comment]
Also what is our timeline for expecting to merge this?

File: autorest/adal/token.go [hunk 1]
```diff
@@ -619,9 +623,24 @@ func (spt *ServicePrincipalToken) refreshInternal(resource string) error {
...
+			http.StatusHTTPVersionNotSupported,
+			http.StatusVariantAlsoNegotiates,
+			http.StatusInsufficientStorage,
+			http.StatusLoopDetected,
+			http.StatusNotExtended,
+			http.StatusNetworkAuthenticationRequired)
+	}
+
+	resp, err := autorest.SendWithSender(spt.sender, req,
+		autorest.DoRetryForStatusCodes(autorest.DefaultRetryAttempts, time.Second*1, retries...),
```
Comment (jhendrixMSFT) [inline autorest/adal/token.go]: Is there no way we can use the retry attempts and delay that the caller configs on the client?

**Comment by mcardosos** [comment]
I have local changes, will push them later today.

**Comment by marstr** [comment]
Closing as stale

**Pull request closed without merging**